react/javascript/src/components/app/OSIcon.tsx

```diff
@@ -0,0 +1,19 @@
+import React, { VoidFunctionComponent } from 'react'
+import { Windows } from './icons/Windows'
+import { Linux } from './icons/Linux'
+import { MacOS } from './icons/MacOS'
+
+export const OSIcon: VoidFunctionComponent<{ name: string }> = ({ name }) => {
+  // TODO need to pattern match more permissively here
```

Comment:
I'm aware this isn't good enough. Are you able to share what's in Cucumber Reports for this?

Reply:
Pretty close. Here:

```tsx
  if (!name) {
    return <span>Unknown OS</span>
  }

  if (name.match(/windows|win32/i)) {
    return <Windows />
  }
  if (name.match(/(darwin|mac)/i)) {
    return <MacOS />
  }
  if (name.match(/linux/i)) {
    return <Linux />
  }

  return <span>{name}</span>
```